pkg/http/handler/timeout_test.go

```diff
@@ -78,14 +81,17 @@ type timeoutHandlerTestScenario struct {
 	name             string
 	firstByteTimeout time.Duration
 	idleTimeout      time.Duration
-	handler          func(mux *sync.Mutex, writeErrors chan error) http.Handler
+	handler          func(clock *clock.FakeClock, mux *sync.Mutex, writeErrors chan error) http.Handler
 	timeoutMessage   string
 	wantStatus       int
 	wantBody         string
 	wantWriteError   bool
 	wantPanic        bool
 }
 
+// This has to be global as the timer cache would otherwise return timers from another clock.
```

Comment:
I'm wondering if we should fix this by making the timer cache non-global, rather than making this global?

Reply:
I did think about that but decided to keep as-is as the current way tests getting and returning timers to the pool a little. If you feel strongly though, we can for sure change things as well.

Reply:
Nope, sounds legit to me